server/tabby_server/api/books.py

```diff
@@ -175,3 +176,90 @@ def _get_result_dict(books: list[google_books.Book]) -> dict:
         "results": book_dicts,
         "resultsCount": results_count,
     }
+
+
+@subapp.route("/recommendations", methods=["GET"])
+def books_recommendations() -> tuple[dict, HTTPStatus]:
+    """Gets a list of recommendations for the given set of books.
+
+    The request must contain parameters for two parallel lists.
+
+    Required args:
+        titles: List with each element being the title of each book, each
+            separated by |---|
+        authors: List with each element being the author(s) of each book, each
+            separated by |---|
+        weights: List of numbers corresponding to how heavily weighed is each
+            book, separated by |---|. Each number is from 0 to 1.
+    """
+
+    # Titles and authors are separated by |---| because they might contain
+    # punctuation like commas, semicolons, etc.
+
+    # Check required args
+    titles_str = request.args.get("titles")
+    if not titles_str:
+        return {
+            "message": "Must specify 'titles' as a non-empty query parameter."
+        }, HTTPStatus.BAD_REQUEST
+    authors_str = request.args.get("authors")
+    if not authors_str:
+        return {
+            "message": "Must specify 'authors' as a non-empty query parameter."
+        }, HTTPStatus.BAD_REQUEST
+    weights_str = request.args.get("weights")
+    if not weights_str:
+        return {
+            "message": "Must specify 'weights' as a non-empty query parameter."
+        }, HTTPStatus.BAD_REQUEST
+
+    # Extract each title and author
+    titles_list = [t.strip() for t in titles_str.split("|---|")]
+    authors_list = [a.strip() for a in authors_str.split("|---|")]
+    if any(t == "" for t in titles_list):
+        return {
+            "message": ("'titles' cannot have an empty element.")
+        }, HTTPStatus.BAD_REQUEST
+    if any(a == "" for a in authors_list):
+        return {
+            "message": ("'authors' cannot have an empty element.")
+        }, HTTPStatus.BAD_REQUEST
+
+    # Extract each weight
+    weights_list = []
+    for i, wstr in enumerate(weights_str.split("|---|")):
+        wstr = wstr.strip()
+        try:
+            w = float(wstr)
+            if not (0.0 <= w <= 1.0):
+                return {
+                    "message": (f"Element {i} at 'weights' is not in [0, 1].")
```

Comment:
I think it would be more accurate to say something like "weights is not within range (0.0 < x < 1.0) or something."